okgo/check/check.go

```diff
@@ -27,6 +27,69 @@ import (
 )
 
 func Run(projectParam okgo.ProjectParam, checkersToRun []okgo.CheckerType, pkgPaths []string, projectDir string, factory okgo.CheckerFactory, parallelism int, stdout io.Writer) error {
+	checkers, maxTypeLen, err := getCheckersToRun(projectParam, checkersToRun, factory)
+	if err != nil {
+		return err
+	}
+	// if there are fewer checkers than max parallelism, update parallelism to number of checkers
+	if len(checkers) < parallelism {
+		parallelism = len(checkers)
+	}
+
+	jobs := make(chan okgo.CheckerParam, len(checkers))
+	results := make(chan checkResult, len(checkers))
+
+	var checksWithFailures []string
+	pullResultsOff := func(toRun int) {
+		for i := 0; i < toRun; i++ {
+			checkResult := <-results
+			if checkResult.producedOutput {
+				checksWithFailures = append(checksWithFailures, string(checkResult.checkerType))
+			}
+		}
+	}
+
+	startASingleWorker := func() {
+		go singleCheckWorker(pkgPaths, projectDir, maxTypeLen, parallelism > 1, jobs, results, stdout)
+	}
+	// We always have 1 worker no matter what
+	startASingleWorker()
+
+	// First run all jobs that need to be ran one at a time
+	checkersWeCannotRunInParallel, checkersWeCanRunInParallel, err := partitionCheckerJobs(checkers)
```

Comment:
Minor semantic naming suggestions/preferences

```suggestion
	// Bucket checks into serial and parallel ones and run all serial ones first
	checkersToRunSerially, checkersToRunInParallel, err := partitionCheckerJobs(checkers)
```

Reply:
Done